content/uc-doc/upgrade/upgrade_notes_details/22-16/provd_plugins_python3.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Provisioning Plugins Migration to Python 3 Upgrade Notes
+---
+
+You might be impacted by the migration of wazo-provd and the provisioning plugins to Python 3 if:
+
+- You have old plugins that are no longer provided by us installed
+- You are using another plugin repository
+
+In those cases, you may have one or many error messages in the wazo-provd logs that look a bit like
+this:
+
+```
+2022-12-16 11:40:11,370 [3813] (ERROR) (provd.app): Error while loading plugin xivo-polycom-4.0.9
+  File "/usr/lib/python3/dist-packages/provd/app.py", line 816, in _pg_load
+    self.pg_mgr.load(plugin_id, gen_cfg, spec_cfg)
+  File "/usr/lib/python3/dist-packages/provd/plugins.py", line 1144, in load
+    self._execplugin(plugin_dir, plugin_globals)
+  File "/usr/lib/python3/dist-packages/provd/plugins.py", line 1047, in _execplugin
+    exec(compile(f.read(), entry_file, 'exec'), pg_globals)
+  File "/var/lib/wazo-provd/plugins/xivo-polycom-4.0.9/entry.py", line 19, in <module>
+    execfile_('common.py', common_globals)
+  File "/usr/lib/python3/dist-packages/provd/plugins.py", line 1039, in aux
+    exec(compile(f.read(), filename, 'exec'), globals, *args, **kwargs)
+  File "/var/lib/wazo-provd/plugins/xivo-polycom-4.0.9/common.py", line 206
+    except tzinform.TimezoneNotFoundError, e:
+                                         ^
+SyntaxError: invalid syntax
+```
+
+In this case, the plugin is no longer provided by Wazo and is still using Python 2 syntax. The only
+possible fixes are:
+
+- Remove the affected plugin
+- Manually update the plugin code to fix the issues
+- Replace the plugin used by the affected devices by a newer version of the plugin (if available)
+
+## Remove the affected plugins
+
+If you decide to actually remove the plugin, you can use `wazo-provd-cli` to do so. The CLI comes
+with helpers that makes it easy to do such tasks.
+
+```shell
+root@stack:~# wazo-provd-cli
+wazo-provd-cli> plugins.uninstall('xivo-polycom-4.0.9')
+```
+
+## Manually update the plugin code to fix the issues
+
+One of the possible way to fix the loading issue is to look at what the Traceback is about and to
+fix it. In our example, it is a simple SyntaxError. The correct syntax is:
+
+```python
+except tzinform.TimezoneNotFoundError as e:
+```
+
```

Comment:
Awesome guide! A few other things have changed however too a side from the standard byte issues. For example the constants for Association are now an Enum eg. `DeviceSupport.COMPLETE`. Perhaps we could mention it or simple say to check our guide or en example plugin?
https://wazo-platform.org/uc-doc/contributors/provisioning/developing_plugins